### PR TITLE
[certificates] Retirement certificates via api

### LINKED ISSUE #603: [pdfs] Move PDF generation to nextjs api endpoint
https://github.com/KlimaDAO/klimadao/pull/586#pullrequestreview-1073219917

### LINKED ISSUE #802: [retirement] PDF receipt not rendering properly
As reported by Dionysus, the PDF receipt download is not rendering properly.

Image

Can use this as an example: https://www.klimadao.finance/retirements/brice.klima/8

Also include an actual PDF.

[retirement_8_0xea4d28f8218f44c84c8d7cbae5deac41acf66ced.pdf](https://github.com/KlimaDAO/klimadao/files/10068468/retirement_8_0xea4d28f8218f44c84c8d7cbae5deac41acf66ced.pdf)

### PR DESCRIPTION
**Pull request by 0xAeterno**
## Description
Bye jspdf 👋🥹

- Retirement certificate pdfs are now generated via a nextjs api endpoint
- `/api/retirements/<beneficiary_address>/<retirement_index>`
- Removes jspdf and assets from `/public` that were only used for pdf generation
- Base64 encode all assets required for pdf generation (fonts & images)

## Related Ticket

Resolves #802
Resolves #603 

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,5 +26,9 @@
     "extract-strings": "npm run -w site lingui:extract && npm run -w app lingui:extract",
     "compile-strings": "npm run extract-strings && npm run -w site lingui:compile && npm run -w app lingui:compile",
     "extract-strings:dev": "npm run -w site lingui:extract:dev && npm run -w app lingui:extract:dev"
+  },
+  "dependencies": {
+    "@types/pdfkit": "^0.12.8",
```
[the remark this answers is not in the document]
Comment (0xAeterno) [reply package.json:31]: thanks bud 😆

hehe just confirming that the proposed changes work in a vercel deployed environment

File: site/lib/retirementCertificates/index.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+type TokenImageMappingKey = keyof typeof tokenImageMap;
+const tokenImageMap = {
+  bct: logoBCT,
+  nct: logoNCT,
+  ubo: logoUBO,
+  nbo: logoNBO,
+  mco2: logoMCO2,
+};
+
```
Comment (0xAeterno) [inline site/lib/retirementCertificates/index.ts:63]:
```suggestion
```

File: site/lib/retirementCertificates/index.ts [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+// const getImageDataURI = async (url: string) => {
+//   const response = await fetch(url);
+//   const arrayBuffer = await response.arrayBuffer();
+//   // const base64 = Buffer.from(new Uint8Array(arrayBuffer)).toString("base64");
+//   // console.log(arrayBuffer);
+
+//   const image = await axios.get(url, { responseType: "arraybuffer" });
+//   // console.log(image.data);
+//   return image.data;
+// };
```
Comment (0xAeterno) [inline site/lib/retirementCertificates/index.ts:72]:
```suggestion
```

**Comment by jabby09** [comment]
@0xAeterno first attempt worked for me! (PDF renders)

The only issue I noticed was we are not wrapping text on long content entries, such as this project name:
Screenshot 2022-12-19 at 9 25 27 AM

**Comment by jabby09** [comment]
@0xAeterno now I am wondering if maybe the issue is that the pdf should be in landscape versus portrait orientation. See the big gap between the description and beneficiary address,
Screenshot 2022-12-19 at 9 29 42 AM

**Comment by Atmosfearful** [comment]
Closing in favor of #1080

**Pull request closed without merging**